src/components/PostDetails.tsx

```diff
@@ -2,105 +1,124 @@
+import React, { useEffect, useState } from 'react';
 import { Loader } from './Loader';
 import { NewCommentForm } from './NewCommentForm';
+import { Post } from '../types/Post';
+import { createComment, deleteComment, getPostComments } from './api/users';
+import { Comment } from '../types/Comment';
+
+type Props = {
+  selectedPost: Post | null;
+};
+
+export const PostDetails: React.FC<Props> = ({ selectedPost }) => {
+  const [comments, setComments] = useState<Comment[]>([]);
+  const [isLoading, setIsLoading] = useState(false);
+  const [isErrorComments, setIsErrorComments] = useState(false);
+  const [isShownForm, setIsShownForm] = useState(false);
+
+  const isNoComments =
+    selectedPost && comments.length === 0 && !isLoading && !isErrorComments;
+
+  useEffect(() => {
+    setIsErrorComments(false);
+    setIsShownForm(false);
+    if (selectedPost) {
+      setIsLoading(true);
+      getPostComments(selectedPost.id)
+        .then(setComments)
+        .catch(() => setIsErrorComments(true))
+        .finally(() => setIsLoading(false));
+    }
+  }, [selectedPost]);
+
+  const addComment = ({ name, email, body, postId }: Omit<Comment, 'id'>) => {
+    return createComment({ name, email, body, postId }).then(newComment =>
+      setComments(currentComments => [...currentComments, newComment]),
+    );
+  };
+
+  const handleDeleteComment = (commentId: number) => {
+    setComments(currentComments =>
+      currentComments.filter(comment => comment.id !== commentId),
+    );
+    deleteComment(commentId);
```

Comment:
You're directly manipulating the state by deleting the comment from the comments array and then calling the `deleteComment` function. It's better to first delete the comment from the server and then update the state based on the response from the server. This ensures that the state always reflects the actual data on the server.